.github/workflows/daily.yml

```diff
@@ -0,0 +1,375 @@
+    steps:
+      - name: Checkout pydantic
+        uses: actions/checkout@v3
+        with:
+          repository: pydantic/pydantic
+      - name: Checkout typing_extensions
+        uses: actions/checkout@v3
+        with:
+          path: typing-extensions-latest
+      - name: Setup pdm for pydantic tests
+        uses: pdm-project/setup-pdm@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: true
+      - name: Add local version of typing_extensions as a dependency
+        run: pdm add ./typing-extensions-latest
+      - name: Install pydantic test dependencies
+        run: pdm install -G testing -G email
+      - name: List installed dependencies
+        run: pdm list -vv  # pdm equivalent to `pip list`
+      - name: Run pydantic tests
+        run: pdm run pytest
+
+  typing_inspect:
+    name: typing_inspect tests
+    if: >-
+      # if 'schedule' was the trigger,
+      # don't run it on contributors' forks
+      ${{
+        github.repository == 'python/typing_extensions'
+        || github.event_name != 'schedule'
+      }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout typing_inspect
+        uses: actions/checkout@v3
+        with:
+          repository: ilevkivskyi/typing_inspect
+      - name: Checkout typing_extensions
+        uses: actions/checkout@v3
+        with:
+          path: typing-extensions-latest
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install typing_inspect test dependencies
+        run: pip install -r test-requirements.txt
+      - name: Install typing_extensions latest
+        run: pip install ./typing-extensions-latest
+      - name: List all installed dependencies
+        run: pip freeze --all
+      - name: Run typing_inspect tests
+        run: pytest
+
+  pyanalyze:
+    name: pyanalyze tests
+    if: >-
+      # if 'schedule' was the trigger,
+      # don't run it on contributors' forks
+      ${{
+        github.repository == 'python/typing_extensions'
+        || github.event_name != 'schedule'
+      }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out pyanalyze
+        uses: actions/checkout@v3
+        with:
+          repository: quora/pyanalyze
+      - name: Checkout typing_extensions
+        uses: actions/checkout@v3
+        with:
+          path: typing-extensions-latest
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install pyanalyze test requirements
+        run: |
+          pip install .
+          pip install pytest mypy_extensions attrs
```

Comment:
let me add an extra into pyanalyze so we can just do `pip install .[test]` or something here